packages/schema/lib/functional-constraints/AuthFieldisSafe.js

```diff
@@ -0,0 +1,78 @@
+'use strict';
+
+const jsonschema = require('jsonschema');
+
+const AUTH_FIELD_ID = '/AuthFieldSchema';
+const AUTH_FIELDS_ID = '/AuthFieldsSchema';
+
+const FORBIDDEN_KEYS = [
+  'access_token',
+  'api_key',
+  'apikey',
+  'api-key',
+  'auth',
+  'jwt',
+  'passwd',
+  'password',
+  'pswd',
+  'refresh_token',
+  'secret',
+  'set-cookie',
+  'signature',
+  'token',
+];
+
+const isSensitiveKey = (key = '') =>
+  FORBIDDEN_KEYS.some((forbidden) =>
+    key.toLowerCase().includes(forbidden.toLowerCase()),
+  );
+
+const checkAuthField = (field) => {
+  const errors = [];
+
+  // if the field key contains any forbidden substring (case-insensitive),
+  // AND 'isSecret' is true, throw a validation error
+  if (isSensitiveKey(field.key) && field.isNotSecret === true) {
```

Comment:
I would also prevent fields with type `password` to be marked as no secret:

```suggestion
  if ((field.key === 'password' || isSensitiveKey(field.key)) && field.isNotSecret === true) {
```